### DISCUSSION TITLE
QUESTIONING ABOUT YOLO MODEL SIZE (small, medium, large)

### DISCUSSION THREAD

**Discussion by sonnam2003** [ultralytics/ultralytics#22595] [Q&A]
Dear everyone, 

I would like to as when selecting a model size (e.g., YOLOv5s, YOLOv5m, YOLOv5l), is the choice primarily influenced by the dataset size, or should it be based on the complexity of the dataset (e.g., fine-grained details, variability in object shapes, background noise, or especially marine debris dataset), or the mAP results we want (e.g., 99% for mAP 50-95)? How should I determine which model size would be most suitable for my object detection task?

Thanks and best regards
<img width="1907" height="941" alt="Screenshot 2025-11-06 100249" src="https://github.com/user-attachments/assets/81469a82-9e0c-4ae4-912a-f714a1f7ed20" />

**Comment by UltralyticsAssistant** [comment]
👋 Hello @sonnam2003, thank you for your interest in Ultralytics 🚀! This is an automated response to help you get started—an Ultralytics engineer will also assist you here soon.

We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.

If this is a custom training ❓ Question (e.g., choosing model size for your dataset), please share more details so we can guide you better:
-Your dataset summary (classes, number of images, example images)
-Task constraints (latency/FPS, GPU/CPU memory limits, target device)
-Your goals (target mAP, throughput, model size preferences)
-Any training logs or results you’ve tried so far
-And confirm you’re following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/)

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):
-**Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
-**Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
-**Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
-**Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Reply by glenn-jocher** [reply]
Great question—choose the smallest model that meets your deployment constraints first (latency/memory), then scale by dataset difficulty: start with YOLO11s as a baseline, increase `imgsz` (e.g., 896–1024) for small/fine objects like marine debris, and only move to YOLO11m/l if accuracy plateaus; targeting 99% mAP50–95 is rarely realistic on real-world data—see the [YOLOv5 vs YOLO11 comparison](https://docs.ultralytics.com/compare/yolov5-vs-yolo11/) for why we recommend YOLO11 for new projects.

```bash
# Baseline, then scale up only if needed
yolo train model=yolo11s.pt data=path/to/data.yaml imgsz=896 epochs=100
yolo train model=yolo11m.pt data=path/to/data.yaml imgsz=896 epochs=100
# Compare results
yolo val model=runs/detect/train/weights/best.pt data=path/to/data.yaml
```

**Reply by sonnam2003** [reply]
Dear **Mr. Glenn**,

Thank you very much for your kind advice. Since I am quite new to AI, I may not have fully understood your explanation, so please allow me to clarify my understanding. Do you mean that, for the first training run, I should always start with the small model size (even when the dataset is relatively large, e.g., around 6k images and quite complex like the marine debris dataset), while increasing the input image size to 896–1024 (instead of the default imgsz=416)? And then if the mAP does not improve, we will start moving to a larger model size?

To provide more context, I am currently working on a project using YOLO11 to **detect marine debris Real-time** with a dataset of approximately 6,000 training images across 3 classes:

Class 0 (Bio): 2,237 samples (20.57%)

Class 1 (ROV): 1,489 samples (13.69%)

Class 2 (Trash): 7,147 samples (65.73%)

for a total of 10,873 bounding boxes.

Also in this project, I am applying transfer learning by freezing all the early layers (backbone + neck) and unfreezing only the final detection head for training.

Given this setup, I would greatly appreciate your advice on how to achieve the highest possible mAP50–95 (ideally around 85–95%). I understand that this dataset is strongly affected by environmental factors such as color variations, object deformation when partially buried by sand, and illumination changes due to depth. I have tried multiple training runs with different hyperparameters (e.g, hsv), but the best result I have achieved so far is around 70% mAP50–95.

Could you please share your recommendations on **the most effective methods or strategies to further improve the mAP50–95 performance** while maintaining real-time deployment with the lowest Latency?

Thank you again for your time and valuable guidance.


*Here is my code:

<img width="677" height="329" alt="Screenshot 2025-11-06 180233" src="https://github.com/user-attachments/assets/df935f1a-d73b-4de1-a65e-b80d57d4a679" />


<img width="394" height="231" alt="Screenshot 2025-11-06 180256" src="https://github.com/user-attachments/assets/188d53e8-a602-4e05-9c98-bdb052982c2e" />

**Reply by glenn-jocher** [reply]
Thanks for the details—yes, start with the smallest model that meets your latency/memory budget (typically YOLO11s) at higher `imgsz` (896–1024) and only step up to YOLO11m/l if val mAP plateaus; for underwater domain shift, avoid head‑only training—either fine‑tune all layers or do a short frozen warm‑up then unfreeze, and consider offline tiling for tiny debris as suggested in our [model evaluation insights](https://docs.ultralytics.com/guides/model-evaluation-insights/).

```bash
# 1) Strong small-model baseline at high resolution
yolo train model=yolo11s.pt data=debris.yaml imgsz=1024 batch=auto epochs=200 patience=50

# 2) Optional: brief head/backbone warm-up, then full fine-tune
yolo train model=yolo11s.pt data=debris.yaml imgsz=1024 epochs=10 freeze=10
yolo train resume=True freeze=0 epochs=200

# 3) If mAP plateaus, scale the model
yolo train model=yolo11m.pt data=debris.yaml imgsz=1024 batch=auto epochs=200 patience=50

# 4) Verify real-time on your device and export for deployment
yolo benchmark model=runs/detect/train/weights/best.pt imgsz=1024 device=0
yolo export model=runs/detect/train/weights/best.pt format=engine half=True
```

Expect 85–95% mAP50–95 to be challenging given visibility/label noise; measure accuracy/latency trade-offs on target hardware with [Benchmark mode](https://docs.ultralytics.com/modes/benchmark/) and deploy via TensorRT using the [Export guide](https://docs.ultralytics.com/modes/export/). If you still stall, please share your `results.csv` and a few representative images here for further pointers.

**Reply by sonnam2003** [reply]
Dear **Mr. Glenn**,

Thank you very much for your valuable advice. I truly appreciate your time and support. I will give it a try and see how it works. If I encounter any difficulties, I will let you know.

Once again, thank you for your helpful suggestion.

Best regards,
Nam

**Reply by glenn-jocher** [reply]
Sounds good Nam, thanks for following up—please feel free to update this discussion with your results or any specific hurdles you hit so others working on similar marine debris use cases can benefit too.

**Reply by sonnam2003** [reply]
Dear **Mr. @glenn-jocher**  ,

I would like to ask which imgsz value I should use for training when my original image resolution is 1920 × 1080 (~300k images at 1920 × 1080 and ~12k images with smaller resolutions such as 953 × 736 or 1199 × 808).

I am considering setting imgsz = 1920 or 1280, as I want the model to capture fine details and achieve the highest possible accuracy.

In addition, I would like to ask whether using an imgsz larger than the recommended 640 × 640 for YOLO models would have any negative impact. (Hardware is not a concern, as I am renting a GPU server for training.)

Thank you for your helpful suggestion.

Best regards,
Nam

**Reply by glenn-jocher** [reply]
Good question Nam—given 1920×1080 inputs, I’d treat 1280 (or at most ~1024–1280) as your practical training `imgsz`: it preserves most detail while avoiding the big hit in memory and speed you’d see at 1920, and using `imgsz` larger than 640 is fully supported with no downside beyond slower training/inference and higher VRAM use. A simple approach is:

```bash
# Try a higher-res baseline and compare accuracy vs speed
yolo train model=yolo11s.pt data=debris.yaml imgsz=1280 batch=auto epochs=200
```

If your smallest debris still look reasonably large (roughly ≥10–15 pixels across) after resizing to 1280, going all the way to 1920 usually brings only marginal mAP gains; if you clearly see underfitting at lower sizes, then increasing `imgsz` or model capacity is a good next step as discussed in the [underfitting guide](https://www.ultralytics.com/glossary/underfitting).
